src/types/NodeType.ts

```diff
@@ -133,16 +154,19 @@ export default function createNodeType({
     return obj[fieldName];
   }
 
-  async function asyncResolve(resolve, obj, args, context, info) {
+  async function asyncResolve(
+    resolve: GraphQLFieldResolver<any, any>,
+    obj: ObjStub,
+    args: Record<string, string>,
+    context: Context,
+    info: GraphQLResolveInfo,
+  ) {
     const fullObj = await getNodeObject(obj, context, info);
     return fullObj && resolve(fullObj, args, context, info);
   }
 
-  function createResolve<K: {}>(
-    resolve: GraphQLFieldResolver<
-      ObjStub & $ObjMap<K, <V>(Class<V>) => ?V>,
```

Comment:
Apparently there's no parallel to `$ObjMap` in typescript so just made it `ObjStub`. Given that it's now a Record type anyways I assumed it was fine

Reply:
I defer to @taion here, this was a Very Clever meta type that helped provide some bonus types based of of the inferred type of the value in this map.

Since TS allows specifying a Generic at the func call site I think we jsut want to template `createResolve` such that `K` is `obj` in the resolver

Reply:
this is busted to say the least, and only works in trivial cases. I'd be ok with dropping createResolve altogether, or simply make it a normal generic function (drop the second param)

Reply:
yeah I'm pretty sure this is only helpful in flow because you can't specify a generic at the function call site

Reply:
agree, super broken in practice

as long as we can make sure that the keys of the objects match with the type, we're good

or else just take the obj value types as `any | null | unknown` and ignore the runtime object value

Reply:
so, can we drop this entirely?

Reply:
If there’s no reason to keep it then let’s drop. I’m a bit hazy on why this was needed in the first place